mso/resource_mso_schema_site_service_graph.go

```diff
@@ -0,0 +1,242 @@
+package mso
+
+import (
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/ciscoecosystem/mso-go-client/client"
+	"github.com/ciscoecosystem/mso-go-client/container"
+	"github.com/ciscoecosystem/mso-go-client/models"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+)
+
+func resourceMSOSchemaSiteServiceGraph() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceMSOSchemaSiteServiceGraphCreate,
+		Read:   resourceMSOSchemaSiteServiceGraphRead,
+		Update: resourceMSOSchemaSiteServiceGraphUpdate,
+		Delete: resourceMSOSchemaSiteServiceGraphDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: resourceMSOSchemaSiteServiceGraphImport,
+		},
+
+		SchemaVersion: version,
+
+		Schema: (map[string]*schema.Schema{
+			"schema_id": &schema.Schema{
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringLenBetween(1, 1000),
+			},
+			"template_name": &schema.Schema{
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringLenBetween(1, 1000),
+			},
+			"site_id": &schema.Schema{
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringLenBetween(1, 1000),
+			},
+			"service_graph_name": &schema.Schema{
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringLenBetween(1, 1000),
+			},
+			"service_node": &schema.Schema{
+				Type:        schema.TypeList,
+				Required:    true,
+				Description: "Configure service nodes for the service graph.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"device_dn": &schema.Schema{
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validation.StringLenBetween(1, 1000),
+						},
+					},
+				},
+			},
+		}),
+	}
+}
+
+func resourceMSOSchemaSiteServiceGraphImport(d *schema.ResourceData, m interface{}) ([]*schema.ResourceData, error) {
+	log.Printf("[DEBUG] %s: Beginning Import", d.Id())
+
+	msoClient := m.(*client.Client)
+	get_attribute := strings.Split(d.Id(), "/")
+	schemaId := get_attribute[0]
+	siteId := get_attribute[2]
+	templateName := get_attribute[4]
+
+	cont, err := msoClient.GetViaURL(fmt.Sprintf("api/v1/schemas/%s", schemaId))
+	if err != nil {
+		return nil, err
+	}
+
+	var graphName string
+	graphName = get_attribute[6]
```

Comment:
not sure why this is not combined
```suggestion
	graphName := get_attribute[6]
```

Reply:
perhaps also move to the other get_attributes above?